Sources/Networking/SessionDelegate.swift

```diff
@@ -92,22 +99,22 @@ class SessionDelegate: NSObject {
     }
 
     private func remove(_ task: URLSessionTask) {
+        lock.lock()
+        defer { lock.unlock() }
+
         guard let url = task.originalRequest?.url else {
```

Comment:
Wouldn't you want to use `task.originalURL` here as well? Also in `task(for:)`, so we only ever call `task.originalRequest?.url` once, and that's in the `SessionDataTask` init. Then I don't think you need to adjust the locks (or at least I don't think it will affect the behavior)